src/libs/Animals/TDynamicSystem.cpp

```diff
@@ -5,8 +5,10 @@
 //--------------------------------------------------------------------
 TDynamicSystem::TDynamicSystem()
 {
-    std::fill(attractors, attractors + MAX_DYNAMIC_OBJECTS, nullptr);
-    std::fill(deflectors, deflectors + MAX_DYNAMIC_OBJECTS, nullptr);
+    attractors.reserve(MAX_DYNAMIC_OBJECTS);
```

Comment:
Again, reserving does not change the size of a vector. Consider using resize instead. And no std::fill will be needed, it will fill it with null pointers.